src/masternode.cpp

```diff
@@ -90,6 +90,15 @@ bool CMasternode::UpdateFromNewBroadcast(CMasternodeBroadcast& mnb)
 //
 arith_uint256 CMasternode::CalculateScore(const uint256& blockHash)
 {
+    if (fDIP0001LockedInAtTip) {
```

Comment:
Is it really required to use DIP0001 activation here? IntantSend is disabled via spork at the moment and I'd assume you won't re-enable InstantSend before enough (nearly all) MNs have upgraded? This is completely independent from DIP0001 activation.

Reply:
Scores/ranks are also used in few other places and specifically, mn payments. IS spork is the last one to enable (or at least that's how we usually do). Payment enforcement spork should be kept disabled until most of MNs and miners are upgraded AND mn winners list is stable/consistent across network. Switching score algo via payment enforcement spork could be dangerous too, these things should be done separately imo.

So the idea was to use the "middle" state of DIP0001 activation (`THRESHOLD_LOCKED_IN`) - we should be pretty confident that majority of miners and MNs are upgraded at that point and it should be safe to switch the algo without being banned by the majority of masternodes and/or hashpower.

Alternatively we could introduce some new spork but this algo upgrade is one way only and we should never need to switch new algo off. Hopefully :)

Reply:
Ah I didn't consider the MN payments. So please ignore what I said :)